packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,255 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  isBuiltinSymbolLike,
+  isPromiseLike,
+  isTypeFlagSet,
+} from '../util';
+
+type Options = [
+  {
+    allowClassInstances: boolean;
+  },
+];
+
+type MessageIds =
+  | 'noStringSpreadInArray'
+  | 'noSpreadInObject'
+  | 'noFunctionSpreadInObject'
+  | 'noClassSpreadInObject';
+
+export default createRule<Options, MessageIds>({
+  name: 'no-misused-spread',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Disallow using the spread operator when it might cause unexpected behavior',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      noStringSpreadInArray:
```

Comment:
[Naming] The rule also flags strings spread in objects:

```suggestion
      noStringSpread:
```

Reply:
hmm, I think it's a bug, it shouldn't flag string spread in objects since it's being handled by TS already. WDYT?

[playground](https://deploy-preview-8509--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.ts&code=MYewdgzgLgBAhjAvDA3jAdJg5HCATLGAXwG4g&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9FETRoHaJHBgAviDVA&tsconfig=&tokens=false)

Reply:
Ah, if TypeScript already handles something, then yeah, we don't need to.

Reply:
Fixed